### PR TITLE
Bump version to 2.0.0b6

### PR DESCRIPTION
**Pull request by ashwin153**

### CODE REVIEW

**Comment by Gobot1234** [comment]
This still isn't ready for release. You'll also need to write a changelog for the beta release

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Versions suffixed with `b*` are in `beta` and can be installed with `pip install --pre betterproto`.
 
+## [2.0.0b6] - 2023-06-05
+
+- Added support for @generated marker [#382](https://github.com/danielgtaylor/python-betterproto/pull/382)
+- Pull down the include_default_values argument to to_json [#405](https://github.com/danielgtaylor/python-betterproto/pull/405)
```
Comment (Gobot1234) [inline CHANGELOG.md:13]:
```suggestion
- Pull down the `include_default_values` argument to `to_json()` [#405](https://github.com/danielgtaylor/python-betterproto/pull/405)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fix typechecker compatiblity checks in server streaming methods [#413](https://github.com/danielgtaylor/python-betterproto/pull/413)
+- Enforce serialize_empty for repeated fields [#417](https://github.com/danielgtaylor/python-betterproto/pull/417)
```
Comment (Gobot1234) [inline CHANGELOG.md:15]:
```suggestion
- Fix "empty-valued" repeated fields not being serialised [#417](https://github.com/danielgtaylor/python-betterproto/pull/417)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Pythonize input_type name in py_input_message [#436](https://github.com/danielgtaylor/python-betterproto/pull/436)
+- Drop support for python3.6 [#444](https://github.com/danielgtaylor/python-betterproto/pull/444)
```
Comment (Gobot1234) [inline CHANGELOG.md:17]: This should be made more obvious (see how the previous release did it)

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fix CI [#456](https://github.com/danielgtaylor/python-betterproto/pull/456)
```
Comment (Gobot1234) [inline CHANGELOG.md:18]:
```suggestion
```
This isn't worth mentioning in a changelog

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Add support for pydantic dataclasses [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
```
Comment (Gobot1234) [inline CHANGELOG.md:19]: This might be worth bringing particular attention to as it's such a requested feature

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fix for [#459](https://github.com/danielgtaylor/python-betterproto/issues/459) (pydantic code gen only) [#460](https://github.com/danielgtaylor/python-betterproto/pull/460)
+- Widen from_dict methods to accept any Mapping [#451](https://github.com/danielgtaylor/python-betterproto/pull/451)
```
Comment (Gobot1234) [inline CHANGELOG.md:21]:
```suggestion
- Widen `from_dict()` to accept any `Mapping` [#451](https://github.com/danielgtaylor/python-betterproto/pull/451)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- Added support for @generated marker [#382](https://github.com/danielgtaylor/python-betterproto/pull/382)
+- Pull down the include_default_values argument to to_json [#405](https://github.com/danielgtaylor/python-betterproto/pull/405)
+- Fix typechecker compatiblity checks in server streaming methods [#413](https://github.com/danielgtaylor/python-betterproto/pull/413)
+- Enforce serialize_empty for repeated fields [#417](https://github.com/danielgtaylor/python-betterproto/pull/417)
+- Pythonize input_type name in py_input_message [#436](https://github.com/danielgtaylor/python-betterproto/pull/436)
+- Drop support for python3.6 [#444](https://github.com/danielgtaylor/python-betterproto/pull/444)
+- Fix CI [#456](https://github.com/danielgtaylor/python-betterproto/pull/456)
+- Add support for pydantic dataclasses [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
+- Fix for [#459](https://github.com/danielgtaylor/python-betterproto/issues/459) (pydantic code gen only) [#460](https://github.com/danielgtaylor/python-betterproto/pull/460)
```
Comment (Gobot1234) [inline CHANGELOG.md:20]: I don't particularly think this is worth mentioning considering it fixes something that was never publicly released but if you want to leave it in that's fine

File: CHANGELOG.md [hunk 8]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Widen from_dict methods to accept any Mapping [#451](https://github.com/danielgtaylor/python-betterproto/pull/451)
+- Fix pydantic_dataclasses reference in README [#474](https://github.com/danielgtaylor/python-betterproto/pull/474)
```
Comment (Gobot1234) [inline CHANGELOG.md:22]: Feels very similar to the 2nd most recent comment

File: CHANGELOG.md [hunk 9]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fix dict encoding for timezone aware datetimes [#468](https://github.com/danielgtaylor/python-betterproto/pull/468)
```
Comment (Gobot1234) [inline CHANGELOG.md:23]:
```suggestion
- Fix `dict` encoding for timezone-aware `datetimes` [#468](https://github.com/danielgtaylor/python-betterproto/pull/468)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Replace pkg_resources with importlib [#462](https://github.com/danielgtaylor/python-betterproto/pull/462)
```
Comment (Gobot1234) [inline CHANGELOG.md:24]:
```suggestion
- Replace `pkg_resources` with `importlib` [#462](https://github.com/danielgtaylor/python-betterproto/pull/462)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fix pydict serialization for optional fields [#495](https://github.com/danielgtaylor/python-betterproto/pull/495)
```
Comment (Gobot1234) [inline CHANGELOG.md:25]:
```suggestion
- Fix `to_pydict` serialization for optional fields [#495](https://github.com/danielgtaylor/python-betterproto/pull/495)
```

**Review by Gobot1234 — CHANGES_REQUESTED** [review]
Apart from the general nits I'd appreciate it if you made separated fixes from additions

**Comment by Gobot1234** [comment]
You also need to add https://github.com/danielgtaylor/python-betterproto/commit/182aedaec4a6cfd2465a2c541e5f2485b0aadb58 and https://github.com/danielgtaylor/python-betterproto/commit/182aedaec4a6cfd2465a2c541e5f2485b0aadb58 to this (sorry)

File: CHANGELOG.md [hunk 12]
```diff
@@ -7,6 +7,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Versions suffixed with `b*` are in `beta` and can be installed with `pip install --pre betterproto`.
 
+## [2.0.0b6] - 2023-06-05
```
Comment (Gobot1234) [inline CHANGELOG.md:10]:
```suggestion
## [2.0.0b6] - 2023-06-25
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -7,6 +7,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- Pass `--python_betterproto_opt=pydantic_dataclasses` to enable code generation of [Pydantic dataclasses](https://docs.pydantic.dev/latest/usage/dataclasses/).
+  Pydantic dataclasses are are drop-in replacement for dataclasses in the standard library that additionally supports validation. For example, a `@validator`
+  is generated for oneof fields that validates that exactly one of the fields is set. Refer to [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
+  and [README.md](https://github.com/danielgtaylor/python-betterproto#generating-pydantic-models) for more information.
+
+- **Breaking**: the minimum Python version has been bumped to `3.7` [#444](https://github.com/danielgtaylor/python-betterproto/pull/444)
+  
+- Added support for @generated marker [#382](https://github.com/danielgtaylor/python-betterproto/pull/382)
```
Comment (Gobot1234) [inline CHANGELOG.md:19]:
```suggestion
- Added support for `@generated` marker [#382](https://github.com/danielgtaylor/python-betterproto/pull/382)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -7,6 +7,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Pull down the `include_default_values` argument to `to_json()` [#405](https://github.com/danielgtaylor/python-betterproto/pull/405)
+- Pythonize input_type name in py_input_message [#436](https://github.com/danielgtaylor/python-betterproto/pull/436)
+- Widen `from_dict()` to accept any `Mapping` [#451](https://github.com/danielgtaylor/python-betterproto/pull/451)
+- Replace `pkg_resources` with `importlib` [#462](https://github.com/danielgtaylor/python-betterproto/pull/462)
+  
+- Fix typechecker compatiblity checks in server streaming methods [#413](https://github.com/danielgtaylor/python-betterproto/pull/413)
+- Fix "empty-valued" repeated fields not being serialised [#417](https://github.com/danielgtaylor/python-betterproto/pull/417)
+- Fix `dict` encoding for timezone-aware `datetimes` [#468](https://github.com/danielgtaylor/python-betterproto/pull/468)
+- Fix `to_pydict` serialization for optional fields [#495](https://github.com/danielgtaylor/python-betterproto/pull/495)
```
Comment (Gobot1234) [inline CHANGELOG.md:28]:
```suggestion
- Fix `to_pydict()` serialization for optional fields [#495](https://github.com/danielgtaylor/python-betterproto/pull/495)
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -7,6 +7,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Versions suffixed with `b*` are in `beta` and can be installed with `pip install --pre betterproto`.
 
+## [2.0.0b6] - 2023-06-05
+
+- Pass `--python_betterproto_opt=pydantic_dataclasses` to enable code generation of [Pydantic dataclasses](https://docs.pydantic.dev/latest/usage/dataclasses/).
+  Pydantic dataclasses are are drop-in replacement for dataclasses in the standard library that additionally supports validation. For example, a `@validator`
+  is generated for oneof fields that validates that exactly one of the fields is set. Refer to [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
+  and [README.md](https://github.com/danielgtaylor/python-betterproto#generating-pydantic-models) for more information.
```
Comment (Gobot1234) [inline CHANGELOG.md:15]:
```suggestion
- Support generating [Pydantic dataclasses](https://docs.pydantic.dev/latest/usage/dataclasses/). 
  Pydantic dataclasses are are drop-in replacement for dataclasses in the standard library that additionally supports validation. Pass `--python_betterproto_opt=pydantic_dataclasses` to enable this feature. Refer to [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
  and [README.md](https://github.com/danielgtaylor/python-betterproto#generating-pydantic-models) for more information.
```

**Review by Gobot1234 — CHANGES_REQUESTED** [review]
Last couple of things

**Review by Gobot1234 — COMMENTED** [review]
Last couple of things

File: CHANGELOG.md [hunk 16]
```diff
@@ -7,6 +7,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Versions suffixed with `b*` are in `beta` and can be installed with `pip install --pre betterproto`.
 
+## [2.0.0b6] - 2023-06-25
+
+- Support generating [Pydantic dataclasses](https://docs.pydantic.dev/latest/usage/dataclasses/). 
+  Pydantic dataclasses are are drop-in replacement for dataclasses in the standard library that additionally supports validation. Pass `--python_betterproto_opt=pydantic_dataclasses` to enable this feature. Refer to [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
+  and [README.md](https://github.com/danielgtaylor/python-betterproto#generating-pydantic-models) for more information.
```
Comment (Gobot1234) [inline CHANGELOG.md:14]:
```suggestion
- Support generating [Pydantic dataclasses](https://docs.pydantic.dev/latest/usage/dataclasses). 
  Pydantic dataclasses are are drop-in replacement for dataclasses in the standard library that additionally supports validation.
  Pass `--python_betterproto_opt=pydantic_dataclasses` to enable this feature.
  Refer to [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
  and [README.md](https://github.com/danielgtaylor/python-betterproto#generating-pydantic-models) for more information.
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -7,6 +7,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+## [2.0.0b6] - 2023-06-25
+
+- Support generating [Pydantic dataclasses](https://docs.pydantic.dev/latest/usage/dataclasses). 
+  Pydantic dataclasses are are drop-in replacement for dataclasses in the standard library that additionally supports validation.
+  Pass `--python_betterproto_opt=pydantic_dataclasses` to enable this feature.
+  Refer to [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
+  and [README.md](https://github.com/danielgtaylor/python-betterproto#generating-pydantic-models) for more information.
+
+- **Breaking**: the minimum Python version has been bumped to `3.7` [#444](https://github.com/danielgtaylor/python-betterproto/pull/444)
```
Comment (Gobot1234) [inline CHANGELOG.md:18]:
```suggestion
- **Breaking**: the minimum Python version has been bumped to `3.7` [#444](https://github.com/danielgtaylor/python-betterproto/pull/444)

- Support generating [Pydantic dataclasses](https://docs.pydantic.dev/latest/usage/dataclasses). 
  Pydantic dataclasses are are drop-in replacement for dataclasses in the standard library that additionally supports validation.
  Pass `--python_betterproto_opt=pydantic_dataclasses` to enable this feature.
  Refer to [#406](https://github.com/danielgtaylor/python-betterproto/pull/406)
  and [README.md](https://github.com/danielgtaylor/python-betterproto#generating-pydantic-models) for more information.
```

**Review by Gobot1234 — APPROVED** [review]

**Comment by Gobot1234** [comment]
Thank you for your work on this

**Comment by MicaelJarniac** [comment]
Will this be released on PyPI? The latest there seems to still be `2.0.0b5`.

**Comment by Gobot1234** [comment]
The release action is currently failing cause @danielgtaylor's email isn't verified on pypi. We are working to remedy the situation

**Comment by danielgtaylor** [comment]
@Gobot1234 I can take a look, sorry about that.

**Comment by MicaelJarniac** [comment]
I see it's now released there. Thanks a lot!
https://pypi.org/project/betterproto/2.0.0b6/

**Comment by ryaminal** [comment]
this is excellent. thanks to @ashwin153 and @Gobot1234 for the effort to get this out.

**Pull request merged**